### ISSUE TITLE
[BUG] Unable to resolve reference $polymer/polymer

### ISSUE THREAD

**Issue by Artur-** [npm/cli#6028]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### This issue exists in the latest npm version

- [X] I am using the latest npm

### Current Behavior

With a given package.json AND package-lock.json

```
% npm install
npm ERR! Unable to resolve reference $@polymer/polymer
```

```
49 verbose stack Error: Unable to resolve reference $@polymer/polymer
49 verbose stack     at get spec [as spec] (/Users/artur/.nvm/versions/node/v19.3.0/lib/node_modules/npm/node_modules/@npmcli/arborist/lib/edge.js:186:15)
49 verbose stack     at OverrideSet.getEdgeRule (/Users/artur/.nvm/versions/node/v19.3.0/lib/node_modules/npm/node_modules/@npmcli/arborist/lib/override-set.js:53:34)
49 verbose stack     at Edge.reload (/Users/artur/.nvm/versions/node/v19.3.0/lib/node_modules/npm/node_modules/@npmcli/arborist/lib/edge.js:228:46)
49 verbose stack     at [_setFrom] (/Users/artur/.nvm/versions/node/v19.3.0/lib/node_modules/npm/node_modules/@npmcli/arborist/lib/edge.js:266:10)
49 verbose stack     at new Edge (/Users/artur/.nvm/versions/node/v19.3.0/lib/node_modules/npm/node_modules/@npmcli/arborist/lib/edge.js:85:19)
49 verbose stack     at [_loadDepType] (/Users/artur/.nvm/versions/node/v19.3.0/lib/node_modules/npm/node_modules/@npmcli/arborist/lib/node.js:890:9)
49 verbose stack     at [Arborist.Node._loadDeps] (/Users/artur/.nvm/versions/node/v19.3.0/lib/node_modules/npm/node_modules/@npmcli/arborist/lib/node.js:867:23)
49 verbose stack     at new Node (/Users/artur/.nvm/versions/node/v19.3.0/lib/node_modules/npm/node_modules/@npmcli/arborist/lib/node.js:259:20)
49 verbose stack     at <computed>.then.error.requiredBy (/Users/artur/.nvm/versions/node/v19.3.0/lib/node_modules/npm/node_modules/@npmcli/arborist/lib/arborist/build-ideal-tree.js:1250:20)
49 verbose stack     at async [nodeFromEdge] (/Users/artur/.nvm/versions/node/v19.3.0/lib/node_modules/npm/node_modules/@npmcli/arborist/lib/arborist/build-ideal-tree.js:1058:19)
50 verbose cwd /Users/artur/Downloads/failing-vaadin24/b
51 verbose Darwin 21.6.0
52 verbose node v19.3.0
53 verbose npm  v9.2.0
54 error Unable to resolve reference $@polymer/polymer
```

### Expected Behavior

npm 8 works without issues

### Steps To Reproduce

```
git clone https://github.com/Artur-/npm-install-fail.git   
cd npm-install-fail
npx npm@9.2.0 i
```

### Environment

- npm: 9.2.0
- Node.js: v19.3.0
- OS Name: mac os
- System Model Name: Macbook M1

**Comment by Artur-** [comment]
9.0.0-pre.5 works fine
9.0.0-pre.6 fails

**Comment by Artur-** [comment]
The relevant changes seem to be
```
commit 2008ea6a807acbd97912799adfe97f276202cea6 (HEAD)
Author: Luke Karrys <luke@lukekarrys.com>
Date:   Tue Oct 18 16:00:26 2022 -0700

    deps: npm-package-arg@10.0.0, pacote@15.0.2

commit 1afe5ba9647d1f0f55bf0a4bace543965d05daed
Author: Gar <gar+gh@danger.computer>
Date:   Tue Oct 11 12:53:51 2022 -0700

    fix: account for new npm-package-arg behavior
    
    `npm`, `npm@`, and `npm@*` are all now the same spec

commit de2d33f3ed42e187803bdd31db4f7a12f08f353c
Author: Nathan Fritz <fritzy@github.com>
Date:   Tue Oct 18 15:47:02 2022 -0700

    feat: add --install-strategy=hoisted|nested|shallow, deprecate --global-style, --legacy-bundling (#5709)
    
    BREAKING CHANGE: deprecate boolean install flags in favor of `--install-strategy`
    * deprecate --global-style, --global now sets --install-strategy=shallow
    * deprecate --legacy-bundling, now sets --install-strategy=nested
```

From commit 2008ea6a807acbd97912799adfe97f276202cea6 by @lukekarrys onward, install fails with
```
npm ERR! Unable to resolve reference $@polymer/polymer
```

Commit 1afe5ba9647d1f0f55bf0a4bace543965d05daed by @wraithgar is the first one causing problems but that one fails with
```
npm ERR! Invalid comparator: latest

35 verbose stack TypeError: Invalid comparator: latest
35 verbose stack     at Comparator.parse (/.../node_modules/semver/classes/comparator.js:38:13)
35 verbose stack     at new Comparator (/.../node_modules/semver/classes/comparator.js:22:10)
35 verbose stack     at /.../node_modules/semver/classes/range.js:133:47
35 verbose stack     at Array.map (<anonymous>)
35 verbose stack     at Range.parseRange (/.../node_modules/semver/classes/range.js:133:35)
35 verbose stack     at /.../node_modules/semver/classes/range.js:34:22
35 verbose stack     at Array.map (<anonymous>)
35 verbose stack     at new Range (/.../node_modules/semver/classes/range.js:34:8)
35 verbose stack     at Object.intersects (/.../node_modules/semver/ranges/intersects.js:4:8)
35 verbose stack     at OverrideSet.getEdgeRule (/.../workspaces/arborist/lib/override-set.js:53:18)

```


Commit de2d33f3ed42e187803bdd31db4f7a12f08f353c works like npm 8

**Comment by eryue0220** [comment]
I wanna try to fix it. can assign to me ?

**Comment by ParadoxInfinite** [comment]
@Artur- I think the issue is fixed in v9.3.0, can you check and confirm? I could only reproduce your issue with your `package-lock.json` When I only used your `package.json` but with a newly generated `package-lock.json` it was working as expected. Anyway, if this is fixed in v9.3.0 for you, do close the issue as resolved. If not, share more info with 9.3.0 and I'll check it out!

**Comment by Artur-** [comment]
The issues reproduces with 9.2 only when the given `package-lock.json` is present and as you say, it works with 9.2 with a newly generated package-lock file. 

Now everything seems ok and I can confirm that 9.3.0 works correctly.
